client-src/elements/chromedash-feature-page.ts

```diff
@@ -119,6 +125,53 @@ export class ChromedashFeaturePage extends LitElement {
     return this.feature && Object.keys(this.feature).length !== 0;
   }
 
+  isFeatureUpcoming(latestStableVersion: number, stages: Array<StageDict>) {
+    if (!latestStableVersion || !stages) {
+      return;
+    }
+
+    const shippingMilestones = new Set<number | undefined>();
+    // Get milestones from all shipping stages, STAGE_TYPES_SHIPPING.
+    // TODO: Do we have to check stage.extensions?
+    for (const stage of stages) {
+      if (STAGE_TYPES_SHIPPING.has(stage.stage_type)) {
+        shippingMilestones.add(stage.desktop_first);
+        shippingMilestones.add(stage.android_first);
+        shippingMilestones.add(stage.ios_first);
+        shippingMilestones.add(stage.webview_first);
+      }
+    }
+    // Check if this feature is shipped within two milestones.
+    if (
+      shippingMilestones.has(latestStableVersion + 1) ||
+      shippingMilestones.has(latestStableVersion + 2)
+    ) {
+      this.isUpcoming = true;
+    }
+  }
+
+  /**
+   * A feature is outdated if it is scheduled to ship in the next 2 milestones,
+   * and its accurate_as_of date is at least 4 weeks ago.*/
+  isFeatureOutdated(): boolean {
+    if (!this.isUpcoming) {
+      return false;
+    }
+
+    if (!this.feature.accurate_as_of) {
+      return true;
+    }
+
+    const accurateDate = Date.parse(this.feature.accurate_as_of);
+    // 4-week period.
+    const gracePeriod = 4 * 7 * 24 * 60 * 60 * 1000;
+    if (accurateDate + gracePeriod < this.currentDate) {
+      return true;
+    }
```

Comment:
Done. I declared a constant variable for ACCURACY_GRACE_PERIOD and left comments on both places. Hope this will remind us to keep it consistent when changed.